app/views/govuk_publishing_components/components/_single_page_notification_button.html.erb

```diff
@@ -1,25 +1,37 @@
 <%
   add_gem_component_stylesheet("single-page-notification-button")
 
-  component_helper = GovukPublishingComponents::Presenters::SinglePageNotificationButtonHelper.new(local_assigns)
+  spnb_helper = GovukPublishingComponents::Presenters::SinglePageNotificationButtonHelper.new(local_assigns)
   shared_helper = GovukPublishingComponents::Presenters::SharedHelper.new(local_assigns)
+  component_helper = GovukPublishingComponents::Presenters::ComponentWrapperHelper.new(local_assigns)
 
   wrapper_classes = %w(govuk-!-display-none-print)
   wrapper_classes << shared_helper.get_margin_bottom
+
+  ga4_data_attributes = ga4_data_attributes ||= nil
+  ga4_link_data_attributes = ga4_data_attributes[:ga4_link] if ga4_data_attributes
+  ga4_link_data_attributes[:url] = spnb_helper.form_action if ga4_link_data_attributes
+
+  component_helper.add_class(wrapper_classes.join(" "))
+  component_helper.add_data_attribute({ module: "gem-track-click" })
+  component_helper.add_data_attribute({ module: ga4_data_attributes[:module] }) if ga4_data_attributes
```

Comment:
This looks good but I think it would help to have an entry in the doc file to demonstrate this functionality.

I created one myself and noticed something odd - the GA4 data module is being added to both the wrapping element and the form, although I'm not quite sure how.